src/ol-ext/format/xsdattribute.js

```diff
@@ -0,0 +1,146 @@
+goog.provide('ngeo.format.XSDAttribute');
+
+goog.require('ol.format.XML');
+
+
+/**
+ * @classdesc
+ * Reads attributes that are defined in XSD format and return them as a list.
+ *
+ * @constructor
+ * @extends {ol.format.XML}
+ * @export
+ */
+ngeo.format.XSDAttribute = function() {
+  goog.base(this);
+};
+goog.inherits(ngeo.format.XSDAttribute, ol.format.XML);
+
+
+/**
+ * @param {Document|Node|string} source Source.
+ * @return {Array.<ngeox.Attribute>} The parsed result.
+ */
+ngeo.format.XSDAttribute.prototype.read = function(source) {
+  return /** @type {Array.<ngeox.Attribute>} */ (
+    goog.base(this, 'read', source)
+  );
+};
+
+
+/**
+ * @param {Document} doc Document.
+ * @return {Array.<ngeox.Attribute>} List of attributes.
+ */
+ngeo.format.XSDAttribute.prototype.readFromDocument = function(doc) {
+  goog.asserts.assert(doc.nodeType == goog.dom.NodeType.DOCUMENT,
+      'doc.nodeType should be DOCUMENT');
+  for (var n = doc.firstChild; n; n = n.nextSibling) {
+    if (n.nodeType == goog.dom.NodeType.ELEMENT) {
+      return this.readFromNode(n);
+    }
+  }
+  return null;
+};
+
+
+/**
+ * @param {Node} node Node.
+ * @return {Array.<ngeox.Attribute>} List of attributes.
+ */
+ngeo.format.XSDAttribute.prototype.readFromNode = function(node) {
+  goog.asserts.assert(node.nodeType == goog.dom.NodeType.ELEMENT,
+      'node.nodeType should be ELEMENT');
+  goog.asserts.assert(node.localName == 'schema',
+      'localName should be schema');
+
+  var elements = node.getElementsByTagName('element');
+  var attributes = [];
+
+  var attribute;
+  for (var i = 0, ii = elements.length; i < ii; i++) {
+    attribute = this.readFromElementNode_(elements[i]);
+    if (attribute) {
+      attributes.push(attribute);
+    }
+  }
+
+  return attributes;
+};
+
+
+/**
+ * @param {Node} node Node.
+ * @return {?ngeox.Attribute} An attribute object.
+ * @private
+ */
+ngeo.format.XSDAttribute.prototype.readFromElementNode_ = function(node) {
+
+  var name = node.getAttribute('name');
+  goog.asserts.assert(name, 'name should be defined in element node.');
+
+  var nillable = node.getAttribute('nillable');
+  var required = !(nillable === true || nillable === 'true');
+
+  var attribute = {
+    name: name,
+    required: required
+  };
+
+  var type = node.getAttribute('type');
+  if (type) {
```

Comment:
The type is required (in `ngeox.Attribute`); please remove this test and add a `goog.asserts.assert` just like you did for `name`